velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -103,12 +193,13 @@ std::vector<std::string> HiveDataSink::finish() const {
 
 void HiveDataSink::close() {
   for (const auto& writer : writers_) {
-    writer->close();
+    if (writer) {
+      writer->close();
+    }
   }
 }
 
-void HiveDataSink::createWriter(
-    const std::optional<std::string>& partitionName) {
+void HiveDataSink::createWriter(vector_size_t partitionId) {
```

Comment:
Any particular reason not to pass partition name along with partition Id?

I noticed that you add input_ as a member variable, but this should not be necessary. The caller can compute partition name and pass it in.

Reply:
input_ member variable was removed. And moved partitionName definition out of createWriter()